### ISSUE TITLE
Secure Keyboard Entry

### ISSUE THREAD

**Issue by koekeishiya** [koekeishiya/khd#7]
![ezw9u](https://cloud.githubusercontent.com/assets/6175959/18895706/f024288a-851b-11e6-8901-840545c09f8e.png)

Picture taken from [here](http://security.stackexchange.com/questions/47749/how-secure-is-secure-keyboard-entry-in-mac-os-xs-terminal)

Khd will not run if Secure Keyboard Entry is enabled, and print an error to stdout/stderr.
It will not record key-events if Secure Keyboard Entry is enabled after Khd has been launched.

It is possible to get around this by working directly with the HID Devices, but I personally do not like that, and have purposely made this decision.

**Comment by beauby** [comment]
Does this mean any binary running with user privilege can see my passwords?

**Comment by tbodt** [comment]
@beauby Yes, if you have given the program permission to use the accessibility api. That's why you have to enter your password to allow a program to use the accessibility api.

**Comment by koekeishiya** [comment]
> Does this mean any binary running with user privilege can see my passwords?

Password prompts can not be monitored by quartz event services, regardless of your "Secure keyboard entry" setting.

Applications that work with raw access to HID devices however, can.

**Comment by L3rchal** [comment]
Strange.... I have "Secure keybord entry" unchecked but still I am unable to start khd due to ```Khd: Secure keyboard entry is enabled! Terminating.```

**Comment by russmatney** [comment]
Thank you for this - was tearing my hair out trying to get iTerm2 to stop capturing all of `khd`'s keybindings, finally found a similar setting via iTerm's dropdown.
